cpp/tests/lists/drop_list_duplicates_tests.cpp

```diff
@@ -14,174 +14,232 @@
+    }
+  }
+  EXPECT_TRUE(results_expected.size() == results.size() && NaN_count == num_NaNs);
 }
 
-TEST_F(DropListDuplicatesTest, FloatingPointTestsNonNull)
+TEST_F(DropListDuplicatesTest, FloatingPointTestsWithNaN)
+{
+  std::vector<float_type> h_input{
+    0, -1, 1, p_NaN, 2, 0, m_NaN, 1, -2, 2, 0, 1, 2, m_NaN, p_NaN, p_NaN, p_NaN, m_NaN};
+  std::unordered_set<float_type> results_expected{-2, -1, 0, 1, 2};
+  test_floating_point(h_input, results_expected);
+}
+
+TEST_F(DropListDuplicatesTest, FloatingPointTestsWithInfAndNaN)
+{
+  std::vector<float_type> h_input{m_inf, 0, m_NaN, 1,     -1,    -2,    p_NaN, p_NaN, p_inf, p_NaN,
+                                  m_NaN, 2, -1,    0,     m_NaN, 1,     2,     p_inf, 0,     1,
+                                  m_inf, 2, m_NaN, p_inf, m_NaN, m_NaN, p_NaN, m_inf};
+  std::unordered_set<float_type> results_expected{-2, -1, 0, 1, 2, m_inf, p_inf};
+  test_floating_point(h_input, results_expected);
+}
+
+TEST_F(DropListDuplicatesTest, StringTestsNonNull)
 {
   // Trivial cases
-  test_once<false>(FLT_LCW{{}}, FLT_LCW{{}});
-  test_once<false>(FLT_LCW{{0, 1, 2, 3, 4, 5}, {}}, FLT_LCW{{0, 1, 2, 3, 4, 5}, {}});
+  test_once(LIST_COL_STR{{}}, LIST_COL_STR{{}});
+  test_once(LIST_COL_STR{"this", "is", "a", "string"}, LIST_COL_STR{"a", "is", "string", "this"});
 
-  // Multiple empty lists
-  test_once<false>(FLT_LCW{{}, {}, {5, 4, 3, 2, 1, 0}, {}, {6}, {}},
-                   FLT_LCW{{}, {}, {0, 1, 2, 3, 4, 5}, {}, {6}, {}});
+  // One list column
+  test_once(LIST_COL_STR{"this", "is", "is", "is", "a", "string", "string"},
+            LIST_COL_STR{"a", "is", "string", "this"});
 
-  auto constexpr p_inf = std::numeric_limits<float_type>::infinity();
-  auto constexpr m_inf = -std::numeric_limits<float_type>::infinity();
+  // Multiple lists column
+  test_once(
+    LIST_COL_STR{LIST_COL_STR{"this", "is", "a", "no duplicate", "string"},
+                 LIST_COL_STR{"this", "is", "is", "a", "one duplicate", "string"},
+                 LIST_COL_STR{"this", "is", "is", "is", "a", "two duplicates", "string"},
+                 LIST_COL_STR{"this", "is", "is", "is", "is", "a", "three duplicates", "string"}},
+    LIST_COL_STR{LIST_COL_STR{"a", "is", "no duplicate", "string", "this"},
+                 LIST_COL_STR{"a", "is", "one duplicate", "string", "this"},
+                 LIST_COL_STR{"a", "is", "string", "this", "two duplicates"},
+                 LIST_COL_STR{"a", "is", "string", "this", "three duplicates"}});
+}
 
-  // Lists contain inf
-  // We can't test for lists containing nan because the order of nan is
-  // undefined after sorting
-  test_once<false>(FLT_LCW{0, 1, 2, 0, 1, 2, 0, 1, 2, p_inf, p_inf, p_inf},
-                   FLT_LCW{0, 1, 2, p_inf});
-  test_once<false>(FLT_LCW{p_inf, 0, m_inf, 0, p_inf, 0, m_inf, 0, p_inf, 0, m_inf},
-                   FLT_LCW{m_inf, 0, p_inf});
+TEST_F(DropListDuplicatesTest, StringTestsWithNulls)
+{
+  auto const null = std::string("");
```

Comment:
```suggestion
  auto const empty = std::string("");
```
This is a pet-peeve of mine. An empty string is not the same as a null string.

Reply:
Oh wait. Nevermind. You are using this as a placeholder identifier.